pennylane/pauli/pauli_arithmetic.py

```diff
@@ -965,8 +972,9 @@ def operation(self, wire_order=None):
         wire_order = wire_order or self.wires
         for pw, coeff in self.items():
             pw_op = pw.operation(wire_order=list(wire_order))
-            summands.append(pw_op if coeff == 1 else s_prod(coeff, pw_op))
-        return summands[0] if len(summands) == 1 else qml.sum(*summands)
+            rep = PauliSentence({pw: coeff})
+            summands.append(pw_op if coeff == 1 else SProd(coeff, pw_op, _pauli_rep=rep))
+        return summands[0] if len(summands) == 1 else Sum(*summands, _pauli_rep=self)
```

Comment:
Providing the pauli_rep so it doesnt get redundantely built again upon initialization is a reaaaally nice catch!